### PR TITLE
refactor: extract string operations to StringOperationsHelper

### PR DESCRIPTION
**Pull request by jlaustill**
## Summary

- Extract string concatenation and substring detection logic from CodeGenerator to dedicated StringOperationsHelper class
- Reduces CodeGenerator from 5237 to 5142 lines (-95 lines)
- Includes 19 unit tests with 97.56% line coverage

## Test plan

- [x] All 951 integration tests pass
- [x] All 5400 unit tests pass
- [x] 97.56% line coverage on new helper

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/transpiler/output/codegen/helpers/StringOperationsHelper.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+   *
+   * @param ctx - Expression context to check
+   * @returns Concatenation operands or null if not a string concat
+   */
+  static getStringConcatOperands(
+    ctx: Parser.ExpressionContext,
+  ): IStringConcatOps | null {
+    // Navigate to the additive expression level using ExpressionUnwrapper
+    const add = ExpressionUnwrapper.getAdditiveExpression(ctx);
+    if (!add) return null;
```
[the remark this answers is not in the document]
Comment (jlaustill) [reply src/transpiler/output/codegen/helpers/StringOperationsHelper.ts]: Fixed. Using `add.MINUS().length > 0` to check for subtraction operators instead of `text.includes("-")`. Added regression test for hyphen-in-literal edge case.
